api/api_ethereum.go

```diff
@@ -502,52 +503,218 @@ type EthRPCTransaction struct {
 	S                *hexutil.Big    `json:"s"`
 }
 
+// newEthRPCTransactionFromBlockAndIndex creates an EthRPCTransaction from block and index parameters.
+func newEthRPCTransactionFromBlockAndIndex(b *types.Block, index uint64) (*EthRPCTransaction, error) {
+	txs := b.Transactions()
+	if index >= uint64(len(txs)) {
+		return nil, errors.New("invalid transaction index")
+	}
+	return newEthRPCTransaction(txs[index], b.Hash(), b.NumberU64(), index)
+}
+
+// newEthRPCTransaction creates an EthRPCTransaction from Klaytn transaction.
+func newEthRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber, index uint64) (*EthRPCTransaction, error) {
+	// When an unknown transaction is requested through rpc call,
+	// nil is returned by Klaytn API, and it is handled.
+	if tx == nil {
+		return nil, nil
+	}
+
+	from := getFrom(tx)
+
+	// If to is nil, it is fills with from.
+	to := tx.To()
+	if to == nil {
+		to = &from
+	}
+
+	// If tx is not TxTypeLegacyTransaction, the type is converted to TxTypeLegacyTransaction.
+	// TODO-Klaytn: In the case of Ethereum transaction type,
+	//  it must be returned as it is without converting the type.
+	typeInt := hexutil.Uint64(tx.Type())
+	if types.TxType(typeInt) != types.TxTypeLegacyTransaction {
+		typeInt = hexutil.Uint64(types.TxTypeLegacyTransaction)
+	}
+
+	signature := tx.GetTxInternalData().RawSignatureValues()[0]
+
+	result := &EthRPCTransaction{
+		Type:     typeInt,
+		From:     from,
+		Gas:      hexutil.Uint64(tx.Gas()),
+		GasPrice: (*hexutil.Big)(tx.GasPrice()),
+		Hash:     tx.Hash(),
+		Input:    tx.Data(),
+		Nonce:    hexutil.Uint64(tx.Nonce()),
+		To:       to,
+		Value:    (*hexutil.Big)(tx.Value()),
+		V:        (*hexutil.Big)(signature.V),
+		R:        (*hexutil.Big)(signature.R),
+		S:        (*hexutil.Big)(signature.S),
+	}
+
+	if blockHash != (common.Hash{}) {
+		result.BlockHash = &blockHash
+		result.BlockNumber = (*hexutil.Big)(new(big.Int).SetUint64(blockNumber))
+		result.TransactionIndex = (*hexutil.Uint64)(&index)
+	}
+
+	// TODO-Klaytn: Have to add additional fields for ethereum transaction types.
+
+	return result, nil
+}
+
+// newEthRPCPendingTransaction creates an EthRPCTransaction for pending tx.
+func newEthRPCPendingTransaction(tx *types.Transaction) (*EthRPCTransaction, error) {
+	return newEthRPCTransaction(tx, common.Hash{}, 0, 0)
+}
+
 // GetTransactionByBlockNumberAndIndex returns the transaction for the given block number and index.
 func (api *EthereumAPI) GetTransactionByBlockNumberAndIndex(ctx context.Context, blockNr rpc.BlockNumber, index hexutil.Uint) *EthRPCTransaction {
-	// TODO-Klaytn: Not implemented yet.
+	block, err := api.publicTransactionPoolAPI.b.BlockByNumber(ctx, blockNr)
+	if block != nil && err == nil {
+		ethTx, err := newEthRPCTransactionFromBlockAndIndex(block, uint64(index))
+		if ethTx == nil || err != nil {
+			return nil
+		}
+		return ethTx
+	}
```

Comment:
How about changing these if branches to more simple structures?
```suggestion
	block, err := api.publicTransactionPoolAPI.b.BlockByNumber(ctx, blockNr)
	if block == nil || err != nil {
	        return nil 
	}
	ethTx, err := newEthRPCTransactionFromBlockAndIndex(block, uint64(index))
	if ethTx == nil || err != nil {
	        return nil 
	}
	return ethTx
	}
```

Reply:
That sounds good :) I applied in [this commit](https://github.com/klaytn/klaytn/pull/1087/commits/bf237b7ede566b340b6bbe2a5a85afc9ee009a66).